### PR TITLE
De-duplicate WP Dashboard Errors

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue:

- #5615 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Comment by kuasha420** [comment]
Thank you @jimmymadon , based on your feedback and the follow up discussion on Slack, I've extracted the added logic to a new `WPDashboardReportError` and used it in all WPDashboard components. I've decided to directly import the new component instead of exposing it in the Widget API, as its usage is limited to the WP Dashboard widgets, but let me know if you have any other ideas. 

Cheers.

**Comment by kuasha420** [comment]
Thank you @jimmymadon for the review. As par the follow up discussion. I've updated the PR accordingly. Please have another look. Cheers.

**Review by techanvil — CHANGES_REQUESTED** [review]
Hey @kuasha420, this is generally looking good, there is one aspect I think we should reconsider though, and I've left couple of minor comments regarding the tests. Please see what you think.

File: assets/js/googlesitekit/widgets/components/WPDashboardReportError.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	const errorInstanceID = useInstanceId(
+		WPDashboardReportError,
+		'WPDashboardReportError'
+	);
+
+	const { setValue } = useDispatch( CORE_UI );
+
+	const errorMessage = Array.isArray( error )
+		? error[ 0 ].message
+		: error.message;
```
[the remark this answers is not in the document]
Comment (kuasha420) [reply assets/js/googlesitekit/widgets/components/WPDashboardReportError.js]: @techanvil  Thank you, this seem reasonable, all things considered so I've gone ahead and implemented this and applied your other suggested changes. Let me know if that looks okay. Cheers.

[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/googlesitekit/widgets/components/WPDashboardReportError.js]: Nice one @kuasha420, LGTM!

**Review by techanvil — APPROVED** [review]
LGTM, nice one @kuasha420!

**Pull request merged**